sdk-lib/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt

```diff
@@ -324,18 +324,16 @@ class SdkSynchronizer private constructor(
     }
 
     override fun getMemos(transactionOverview: TransactionOverview): Flow<String> {
-        return when (transactionOverview.isSentTransaction) {
-            true -> {
-                val sentNoteIds = storage.getSentNoteIds(transactionOverview.id)
-
-                sentNoteIds.map { rustBackend.getSentMemoAsUtf8(it) }.filterNotNull()
-            }
-            false -> {
-                val receivedNoteIds = storage.getReceivedNoteIds(transactionOverview.id)
-
-                receivedNoteIds.map { rustBackend.getReceivedMemoAsUtf8(it) }.filterNotNull()
+        return storage.getNoteIds(transactionOverview.id).map {
+            when (transactionOverview.isSentTransaction) {
```

Comment:
I wonder if the `get*MemoAsUtf8` methods really make sense now, given `v_tx_outputs`? Should we just be getting the memos from there?

Reply:
Eventually yes, that would be more efficient because we could do a single database query.  We don't have any Kotlin implementation for dealing with memo bytes so this is the path of least implementation.